src/utils/fetch.ts

```diff
@@ -1,12 +1,35 @@
 import axios from 'axios';
+import {push} from 'connected-react-router';
 
 import {backendURI} from 'config/main';
 
 export const fetch = axios.create({
 	baseURL: backendURI,
 });
 
-fetch.interceptors.response.use(
-	response => response,
-	error => Promise.reject(error.response),
-);
+export const createAxiosResponseInterceptor = () => {
```

Comment:
do you prefer a promise chaining? maybe we try to use async await? it's same generator ideology like sagas